app.rb

```diff
@@ -141,4 +151,82 @@ def list_rentals
       end
     end
   end
+
+  def load_data
+    if File.exist?('student.json')
+      File.foreach('student.json') do |line|
+        load_people_data(line)
+      end
+    end
+
+    if File.exist?('teacher.json')
+      File.foreach('teacher.json') do |line|
+        load_people_data(line)
+      end
+    end
+
+    if File.exist?('book.json')
+      File.foreach('book.json') do |line|
+        s = JSON.parse(line, create_additions: true)
+        @books.push(s)
+      end
+    end
+
+    File.foreach('rental.json') do |line|
+      load_rental_data(line)
+    end
+  end
+
+  def load_people_data(line)
+    s = JSON.parse(line, create_additions: true)
+    @people.push(s)
+  end
+
+  def load_rental_data(line)
+    s = JSON.parse(line)
+    selected_student = ObjectSpace.each_object(Student).select do |obj|
+      obj.name == s['a'][1]['a'][2] && obj.age == s['a'][1]['a'][0]
+    end
+    selected_teacher = ObjectSpace.each_object(Teacher).select do |obj|
+      obj.name == s['a'][1]['a'][2] && obj.age == s['a'][1]['a'][0]
+    end
+    selected_book = ObjectSpace.each_object(Book).select do |obj|
+      obj.title == s['a'][2]['a'][0]
+    end
+
+    begin
+      s1 = Rental.new(s['a'][0], selected_student[0], selected_book[0])
+      @rentals.push(s1)
+    rescue StandardError
+      s1 = Rental.new(s['a'][0], selected_teacher[0], selected_book[0])
+      @rentals.push(s1)
+    end
+  end
+
+  def save_data
+    File.open('student.json', 'w') do |file|
+      @people.each do |people|
+        file.puts JSON.generate(people) if people.type == 'Student'
+      end
+    end
+
+    File.open('teacher.json', 'w') do |file|
+      @people.each do |people|
+        file.puts JSON.generate(people) if people.type == 'Teacher'
+      end
+    end
+
+    File.open('book.json', 'w') do |file|
+      @books.each do |book|
+        file.puts JSON.generate(book)
+      end
+    end
+
+    File.open('rental.json', 'w') do |file|
+      @rentals.each do |rental|
+        file.puts JSON.generate(rental)
+      end
+    end
+  end
 end
+# rubocop:enable Metrics/ClassLength
```

Comment:
- [ ] As the previous reviewer requested, let's kindly make sure Linters are not disable. Linters are great tools that monitors our codebase and flag programmatic offences. An offence here may be the large file size which should have been refactor into different file sizes. It's easier to debug when functionalities are separated into chunks :+1:

Reply:
Alright. NP.